students/mxwllndrsn/lesson04/mailroom_mk2.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+
+#uw python 210
+#lesson 03
+#max anderson
+
+#mailroom.py
+
+import os
+import io
+from datetime import date
+
+donor_list = [
+              ['Andrew Jackson', 10000, 5, 2000],
+              ['Benjamin Franklin', 253000, 2, 117500],
+              ['John Kennedy', 20, 1, 20],
+              ['Abraham Lincoln', 807, 2, 403.5],
+              ['Jim Jones', 918, 1, 918]
+             ]
```

Comment:
from lesson 4 and forward this should be a dictionary. why use a dictionary? because it is VERY efficient and simplifies the code in _this_ particular case.

your donor name is a natural key in our DB; so `donors = {"joe": [100, 200], "tom": [44]}`

which means looking up a donor:
`donor = donors[donor_name]`
and adding to donor
`donor[donor_name.append(donation)`
becomes simple + efficient

.. and because you get to practice dictionaries because they are honestly one of the most common and useful data structures in python 😉